src/OpenTelemetry.Instrumentation.AspNet/AspNetInstrumentationOptions.cs

```diff
@@ -46,11 +46,27 @@ public class AspNetInstrumentationOptions
     /// </summary>
     /// <remarks>
     /// <para><see cref="Activity"/>: the activity being enriched.</para>
-    /// <para>string: the name of the event.</para>
-    /// <para>object: the raw object from which additional information can be extracted to enrich the activity.
-    /// The type of this object depends on the event, which is given by the above parameter.</para>
+    /// <para><see cref="HttpRequest"/>: the HttpRequest object from which additional information can be extracted to enrich the activity.</para>
     /// </remarks>
-    public Action<Activity, string, object> Enrich { get; set; }
+    public Action<Activity, HttpRequest> EnrichWithHttpRequest { get; set; }
```

Comment:
Random idea. Since we are making this change...

Should we consider switching to `HttpContext` or adding it to the signature?

`Action<Activity, HttpContext>`
or
`Action<Activity, HttpContext, HttpRequest>`

I seem to remember users asking for that way back. Off the top of my head the nice thing `HttpContext` gives you is the items bag so you can easily push stuff into your telemetry.

Reply:
if we pass in the Context, we could get rid of request. And if we do this, do we also change ASP.NET Core to match this?

Reply:
> if we pass in the Context, we could get rid of request.

Ya if you want to keep a single delegate we could probably get away with just `Action<Activity, string, HttpContext, Exception?>`. We could also introduce a contract that is expandable like...

```csharp
public class AspNetInstrumentationOptions
{
   public AspNetEnrichAction Enrich {get; set;}
}

public delegate void AspNetInstrumentationEnrichAction(in AspNetEnrichArguments arguments);

public readonly struct AspNetInstrumentationEnrichArguments 
{
    public string EventName {get;}
    public Activity Activity {get;}
    public HttpContext HttpContext {get;}
    public Exception? Exception {get;}
}
```

> And if we do this, do we also change ASP.NET Core to match this?

We could. Less of an issue there though because the ASP.NET Core request & response objects have a property that link back to the HttpContext. I kind of like that design with a dedicated delegate + arguments struct it feels cleaner? 🤷